python/cli/data_platform_workflows_cli/update_bundle.py

```diff
@@ -24,36 +25,45 @@ def get_ubuntu_version(series: str) -> str:
     ).stdout.split(" ")[0]
 
 
-def fetch_latest_revision(charm, charm_channel, series=None) -> int:
-    """Gets the latest charm revision number in channel."""
+def fetch_charm_info_from_store(charm, charm_channel) -> Tuple[Dict, Dict]:
+    """Returns, for a given channel, the necessary charm info from store endpoint."""
     response = requests.get(
-        f"https://api.snapcraft.io/v2/charms/info/{charm}?fields=channel-map"
+        f"https://api.snapcraft.io/v2/charms/info/{charm}?fields=channel-map,default-release&channel={charm_channel}"
```

Comment:
AFAIU from the docs, `default-release` basically returns the latest revision from the default channel (which in itself is not what we want, most charms in our bundles are using `edge` and not the default `stable`). 

What I realized though is that we can specify which channel we want, then if we pass `&channel=14/edge` it will fetch the last revision released in `14/edge` which is what we're looking for! But yes, this isn't mentioned explicitly in the docs so I'll double check

Reply:
Update: store team confirmed that we can rely on default-release and query for a specific channel, like done here.